corelib/src/odometry/OdometryF2M.cpp

```diff
@@ -818,6 +870,28 @@ Transform OdometryF2M::computeTransform(
 						if(maxNewFeatures_ == 0  || added < maxNewFeatures_)
 						{
 							int cameraIndex = iter->second.second.second.second.second;
+							cv::Point3f pt = iter->second.second.second.first;
+							if(!util3d::isFinite(pt))
+							{
+								// get the ray instead
+								float x = iter->second.second.first.pt.x; //subImageWidth should be already removed
+								float y = iter->second.second.first.pt.y;
+								Eigen::Vector3f ray = util3d::projectDepthTo3DRay(
+										lastFrameModels[cameraIndex].imageSize(),
+										x,
+										y,
+										lastFrameModels[cameraIndex].cx(),
+										lastFrameModels[cameraIndex].cy(),
+										lastFrameModels[cameraIndex].fx(),
+										lastFrameModels[cameraIndex].fy());
+								float scaleInf = 0.05 * lastFrameModels[cameraIndex].fx();
```

Comment:
Doublecheck how much this affects the results. Maybe Euroc dataset didn't work well before of this change.

Reply:
The dataset evaluation is using `Vis/DepthAsMask=true` so this is ignored anyway. I still added a parameter called `OdomF2M/InitDepthFactor` to tune 0.05 if needed.